bundler/spec/dependabot/bundler/update_checker/version_resolver_spec.rb

```diff
@@ -130,6 +130,67 @@
         end
       end
 
+      context "with a Bundler v2 version specified", :bundler_v2_only do
+        let(:requirement_string) { "~> 1.4.0" }
+
+        let(:dependency_files) { project_dependency_files("bundler2/bundler_specified") }
+        its([:version]) { is_expected.to eq(Gem::Version.new("1.4.0")) }
+
+        context "attempting to update Bundler" do
+          let(:dependency_name) { "bundler" }
+          let(:requirement_string) { "~> 2.2.0" }
+
+          let(:dependency_files) { project_dependency_files("bundler2/bundler_specified") }
+
+          it "returns nil as resolution returns the bundler version installed by core" do
```

Comment:
Ref: https://github.com/dependabot/dependabot-core/pull/3319/files#diff-8619130a1efd74403d2096bc77c13c862073216cecadb1ae2e00ec4a62070675R34

Reply:
This is a bummer, it doesn't seem like we can currently update bundler at all.

When running a dry-run to update a gemfile with bundler v1 it tries to go to v2 even though the project has been bundled with v1, it looks like it falls through here atm: https://github.com/dependabot/dependabot-core/blob/main/bundler/helpers/v1/lib/functions/version_resolver.rb#L24 because `bundler` isn't listed as one of the gems in the definition.

When running the v2 helpers, `bundler` is listed as a gem but we [return nil if the gem name is `bundler`](https://github.com/dependabot/dependabot-core/pull/3319/files#diff-8619130a1efd74403d2096bc77c13c862073216cecadb1ae2e00ec4a62070675R34), I think this is the best thing for now as the "resolved version" is core's bundler version, not the latest resolvable version.